.github/workflows/c-check.yml

```diff
@@ -0,0 +1,53 @@
+name: C Code Generation and Formatting Check
+
+on:
+  push:
+  pull_request: {}
+  merge_group: {}
+
+jobs:
+  format-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.4"
+          bundler-cache: none
+      - name: Set working directory as safe
+        run: git config --global --add safe.directory $(pwd)
+      - name: Set up permission
```

Comment:
This step causes an error...
(Other jobs are fixed in https://github.com/ruby/rbs/pull/2485.)